dashboard/generate.py

```diff
@@ -131,15 +149,44 @@ def efu_get_version(self, ipaddr, port):
 
             # return " ".join(data.split()[1:4])
             test = "&#60;br /&#62;".join(data.decode("utf-8").split()[1:4])
-            self.dprint(test)
+            logger.debug(test)
             return test
-        except:
-            self.dprint("connection reset (by peer?)")
+        except Exception as e:
+            logger.error(f"connection reset (by peer?): {e}")
             return "connection reset (by peer?)"
```

Comment:
```suggestion
        except OSError as e:
            logger.exception(f"Socket error on {ip}:{port}")
            return e.msg
        except Exception as e:
            logger.exception(f"Error parsing data for {ip}:{port}")
            return e.msg
```

minor suggestion. use logger.exception in exception handlers for the extra exeption info
https://docs.python.org/3/library/logging.html#logging.exception

Reply:
Also appreciated 🙌 
Thank you 🙂